doc/OnlineDocs/working_abstractmodels/data/dataportals.rst

```diff
@@ -143,8 +139,54 @@ loaded:
   data.
 
 Finally, the ``data()`` method provides a generic mechanism for
-accessing the underlying data representation used by :class:`DataPortal
-<ref pyomo.core.base.DataPortal.DataPortal>` objects.
+accessing the underlying data representation used by :class:`~pyomo.environ.DataPortal` objects.
+
+
+Loading Structured Data
+-----------------------
+
+JSON and YAML files are structured data formats that are well-suited for
+data serialization.  These data formats do not represent data in tabular
+format, but instead they directly represent set and parameter values
+with lists and dictionaries:
+
+* **Simple Set**: a list of string or numeric value
+
+* **Indexed Set**: a dictionary that maps an index to a list of string
+  or numeric value
+
+* **Simple Parameter**: a string or numeric value
+
+* **Indexed Parameter**: a dictionary that maps an index to a numeric
+  value
+
+For example, consider the following JSON file:
+
+.. literalinclude:: ../../tests/dataportal/T.json
+    :language: none
+
+The data in this file can be used to load the following model:
+
+.. literalinclude:: ../../tests/dataportal/dataportal_tab_json1.spy
+    :language: python
+
+Note that no ``set`` or ``param`` option needs to be specified when
+loading a ``JSON`` or ``YAML`` file.  All of the set and parameter
+data in the file are loaded by the :class:`DataPortal <ref
+pyomo.core.base.DataPortal.DataPortal>` object, and only the data
```

Comment:
Looks like you changed all of the other `:class:` references in this section to have a slightly different format, did you miss this one?